pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -522,16 +522,30 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, nodeName string) (diskID string
 
 	err = vm.AddDevice(ctx, disk)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	vmDevices, err = vm.Device(ctx)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	devices := vmDevices.SelectByType(disk)
-	return devices.Name(devices[len(devices)-1]), nil
+	newDevice := devices[len(devices)-1]
+
+	deviceUuid := getVirtualDiskUuid(newDevice)
+
+	return devices.Name(devices[len(devices)-1]), deviceUuid, nil
+}
+
+func getVirtualDiskUuid(newDevice types.BaseVirtualDevice) string {
```

Comment:
should return error if uuid cannot be extracted.  this will cause the mount to not work